sklearn/metrics/_classification.py

```diff
@@ -382,6 +388,9 @@ def confusion_matrix(
             cm = cm / cm.sum()
         cm = np.nan_to_num(cm)
 
+    if n_labels == 2 and pos_label != labels[-1]:
+        cm = cm[::-1, ::-1]
```

Comment:
I feel like an actual default value of `None`, and then defaulting to `1` only when binary, and raising if provided and not binary would have been a better choice. But we've done it as you're doing here for other methods, so I'm kinda fine with it.

Reply:
Actually, we have both ways. Here I am implementing as for `precision_recall_fscore`. `roc_curve` has the following:

```
pos_labelint, float, bool or str, default=None
    The label of the positive class. When pos_label=None, if y_true is in {-1, 1} or {0, 1}, pos_label is set to 1, otherwise an error will be raised.
```

Which is what you want I think.

Reply:
yes, plus raise if `y_true` is provided and it's not a binary case here.